muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/transcriptofrecords/rest/TranscriptofRecordsRESTService.java

```diff
@@ -167,9 +178,105 @@ public Response listWorkspaceActivities(
+                    Mandatority mandatority = activity.getMandatority();
+                    if (mandatority != null && mandatority == Mandatority.MANDATORY) {
+                      mandatoryCourseCredits = Integer.sum(units, mandatoryCourseCredits);
+                   }
+                  }
+                  
+                  // Search for finding out course mandaority
+                  SearchProvider searchProvider = getProvider("elastic-search");
+                  
+                  if (searchProvider != null && activity.getId() != null) {
+                    
+                    WorkspaceEntity workspaceEntity = workspaceController.findWorkspaceEntityById(activity.getId());
+                    workspaceIdentifier = workspaceEntity.schoolDataIdentifier();
+                    SearchResult sr = searchProvider.findWorkspace(workspaceIdentifier);
+                    
+                    List<Map<String, Object>> results = sr.getResults();
+                    for (Map<String, Object> result : results) {
+                      
+                      String educationTypeId = (String) result.get("educationTypeIdentifier");
+
+                      Mandatority mandatority = null;
+
+                      if (StringUtils.isNotBlank(educationTypeId)) {
+                        SchoolDataIdentifier educationSubtypeId = SchoolDataIdentifier.fromId((String) result.get("educationSubtypeIdentifier"));
+                        
+                        EducationTypeMapping educationTypeMapping = workspaceEntityController.getEducationTypeMapping();
+                        
+                        mandatority = (educationTypeMapping != null && educationSubtypeId != null) 
+                            ? educationTypeMapping.getMandatority(educationSubtypeId) : null;
+                        
+                      }
+                      if (mandatority != null && mandatority == Mandatority.MANDATORY) {
+                       mandatoryCourseCredits = Integer.sum(units, mandatoryCourseCredits);
+                       
+                       activity.setMandatority(mandatority);
+                      }
+                    }
+                  } 
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    
+    activityInfo.setCompletedCourseCredits(allCourseCredits);
+    activityInfo.setMandatoryCourseCredits(mandatoryCourseCredits);
+    activityInfo.setShowCredits(false);
+    
+    User user = userController.findUserByDataSourceAndIdentifier(studentIdentifier.getDataSource(), studentIdentifier.getIdentifier());
+    
+    // Find student's curriculum to tell whether the score will be shown to the user
+    
+    String curriculumName = guiderController.getCurriculumName(user.getCurriculumIdentifier());
+    
+    if (curriculumName != null && curriculumName.equals("OPS 2021") && (activityInfo.getLineCategory() != null && activityInfo.getLineCategory().equals("Lukio"))) {
```

Comment:
Again, I assume this could be used as the major flag if any credit counts need to be resolved at all.